splice_machine/src/main/java/com/splicemachine/derby/procedures/BaseAdminProcedures.java

```diff
@@ -192,4 +192,56 @@ protected static ExecRow buildExecRow(ResultColumnDescriptor[] columns) throws S
         return template;
     }
 
+    @FunctionalInterface
+    public interface ConsumeWithException<T, T2, E extends Throwable> {
+        void accept(T t, T2 t2) throws E;
+    }
+    @FunctionalInterface
+    public interface ConsumeWithException3<T, T2, T3, E extends Throwable> {
+        void accept(T t, T2 t2, T3 t3) throws E;
+    }
+
+    public static void executeOnAllServers(ConsumeWithException<HostAndPort, Connection, SQLException> function) throws SQLException {
+        List<HostAndPort> servers;
+        try {
+            servers = EngineDriver.driver().getServiceDiscovery().listServers();
+        } catch (IOException e) {
+            throw PublicAPI.wrapStandardException(Exceptions.parseException(e));
+        }
+
+        for (HostAndPort server : servers) {
+            try (Connection connection = RemoteUser.getConnection(server.toString())) {
+                function.accept(server, connection);
+            }
+        }
+    }
+
+    public static void executeOnAllServers(String sql) throws SQLException {
+
+        executeOnAllServers((hostAndPort, connection) -> execute(connection, sql));
+    }
+
+    public static void executeOnAllServers(String sql,
+                                           ConsumeWithException3<HostAndPort, Connection, ResultSet, SQLException> function)
+            throws SQLException {
+
+        executeOnAllServers( (hostAndPort, connection) -> {
+            if(function == null)
+                execute(connection, sql);
+            else {
+                try (Statement stmt = connection.createStatement()) {
+
+                    try (ResultSet rs = stmt.executeQuery(sql)) {
+                        function.accept(hostAndPort, connection, rs);
+                    }
+                }
+            }
+        });
+    }
+
+    public static void execute(Connection connection, String sql) throws SQLException {
+        try (Statement stmt = connection.createStatement()) {
+            stmt.execute("call SYSCS_UTIL.INVALIDATE_DICTIONARY_CACHE()");
```

Comment:
Should this be `stmt.execute(sql);` ?

Reply:
If function is never null in the other call, this function is never used, so remove too?

Reply:
done (both)